### DISCUSSION TITLE
setting routes on connection loading

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [strongswan/strongswan#2129] [Q&A]
I need to set  a route in the container which all connecting clients will use of a connection

something like this

/sbin/route add -net 192.168.x.0/26 dev eth0

I have tried putting it in the updown script but this executing per client (and does not seem to work). I think it would be nice if this is specified somewhere when the connection is made available.

Where can I realise this?

**Comment by tobiasbrunner** [comment]
Use a charon start script (see https://github.com/strongswan/strongswan/discussions/2128#discussioncomment-8639757) or do it in a script that's run when the container starts.

**Reply by f1-outsourcing** [reply]
I think scripts are started from a different user (ipsec)

`route: ioctl 0x890b failed: Operation not permitted`

**Reply by tobiasbrunner** [reply]
If charon is [configured to change user/group and drop capabilities](https://docs.strongswan.org/docs/5.9/install/reducedPrivileges.html), then yes, the start scripts may run under a different user. By the way, not sure if that makes sense in a container that's already isolated.

**Reply by f1-outsourcing** [reply]
Idk, you have to run the container with privileged mode (or linux capabilities). It goes beyond my knowledge what influence this can have on the host when there are exploits.

Is there another way of doing this, so I can keep things as default as possible?

**Reply by tobiasbrunner** [reply]
Not sure what default you refer to, but I'm sure you could run the command in a script that you configure via ENTRYPOINT before it starts the IKE daemon.

**Reply by f1-outsourcing** [reply]
yes of course, but that is a bit crappy solution.
